sdk/servicebus/service-bus/src/serviceBusClient.ts

```diff
@@ -56,23 +57,69 @@ export class ServiceBusClient {
    */
   private _context: ConnectionContext;
 
+  /**
+   * Creates a ServiceBusClient for the Service Bus Namespace represented in the given connection
+   * string.
+   * @param connectionString - Connection string of the form
+   * 'Endpoint=sb://my-servicebus-namespace.servicebus.windows.net/;SharedAccessKeyName=my-SA-name;SharedAccessKey=my-SA-key'
+   * @param options Options to control ways to interact with the
+   * Service Bus Namespace.
+   * @returns ServiceBusClient
+   */
+  constructor(connectionString: string, options?: ServiceBusClientOptions);
+
   /**
    * Instantiates a ServiceBusClient to interact with a Service Bus Namespace.
    *
    * @constructor
-   * @param {ConnectionConfig} config - The connection configuration needed to connect to the
-   * Service Bus Namespace.
-   * @param {TokenCredential} [tokenCredential] - SharedKeyCredential object or your 
+   * @param host - The host name for the Service Bus namespace. This is likely to be similar to
+   * <yournamespace>.servicebus.windows.net
+   * @param credential - SharedKeyCredential object or your
    * credential that implements the TokenCredential interface.
-   * @param {ServiceBusClientOptions} - Options to control ways to interact with the Service Bus
+   * @param options - Options to control ways to interact with the Service Bus
    * Namespace.
    */
-  private constructor(
-    config: ConnectionConfig,
-    credential: SharedKeyCredential | TokenCredential,
+  constructor(host: string, credential: TokenCredential, options?: ServiceBusClientOptions);
+  constructor(
+    hostOrConnectionString: string,
+    credentialOrServiceBusClientOptions?: TokenCredential | ServiceBusClientOptions,
     options?: ServiceBusClientOptions
   ) {
+    let config;
+    let credential;
+    let connectionString;
+
     if (!options) options = {};
+
+    if (hostOrConnectionString == undefined || typeof hostOrConnectionString !== "string") {
+      throw new Error("Input parameter of host or connection string must be a string.");
+    }
+
+    if (!isTokenCredential(credentialOrServiceBusClientOptions)) {
+      // connectionString and options based constructor was invoked
+      connectionString = hostOrConnectionString;
+      config = ConnectionConfig.create(connectionString);
+      config.webSocket = options && options.webSocket;
+      config.webSocketEndpointPath = "$servicebus/websocket";
+      config.webSocketConstructorOptions = options && options.webSocketConstructorOptions;
+
+      // Since connectionstring was passed, create a SharedKeyCredential
+      credential = new SharedKeyCredential(config.sharedAccessKeyName, config.sharedAccessKey);
+
+      ConnectionConfig.validate(config);
+    } else {
+      if (credentialOrServiceBusClientOptions == undefined) {
+        throw new Error("Input parameter token credential must be defined.");
```

Comment:
This code path will not be hit. `isTokenCredential(undefined)` returns falsy

Reply:
! on falsy is true right? It would go here because the if check is `(!isTokenCredential)` - So technically if someone passed in `undefined` for credentials, we'd be catching it here?

Reply:
! on falsy is indeed true. When `credentialOrServiceBusClientOptions` is undefined, `if (!isTokenCredential(credentialOrServiceBusClientOptions))` will return true. Therefore, we wont enter the else block here at line 112

Reply:
Right, mistakenly assumed the utility to check for type of the options - updated the checks